src/hugr/validate.rs

```diff
@@ -158,7 +159,16 @@ impl<'a, 'b> ValidationContext<'a, 'b> {
             }
         }
 
-        // Check operation-specific constraints
+        // Check operation-specific constraints. Firstly that type args are correct
+        // (note that if the Hugr were mutable here, this could be a good time to do
+        // resolve_extension_ops)
+        if let OpType::LeafOp(crate::ops::LeafOp::CustomOp(b)) = op_type {
+            for arg in b.args() {
+                arg.validate(self.extension_registry)
+                    .map_err(|cause| ValidationError::SignatureError { node, cause })?;
+            }
+        }
+        // Secondly that the node has correct children
```

Comment:
Methinks `validate_operation` is not a good name, given there's already other validation of the operation *here in validate_node*. Rather, validate_operation is all about the op's *children*. However `validate_children` is already taken (as a *subset* of those child-validating checks), so the best renaming I can see would be along the lines of
1. `validate_children` -> `validate_children_op_specific` *or something*
2. `validate_operation` -> `validate_children`

If anyone can think of a good name for the *or something* then I'm happy to proceed with that ;-)

Reply:
Uhm. I guess the `OpValidityFlags` are focused on a container's children.
I'd say rename that to `OpContainerFlags` (?) and `ValidateOp::validate_children` to `run_children_validator` ?
(on top of your 2. )

Reply:
Ok, gone with `validate_children` -> `validate_op_children` (my #1, your #2 - the only `validate_children` was that defined in ValidateOp), and `validate_operation` -> `validate_children`.

I've left OpValidityFlags because it does have one member that is not related to children - `non_df_port`. I think we were talking about refactoring non-dataflow-ports more widely, so if `non_df_port` goes as part of that refactor.... (as perhaps it should: `non_df_port` doesn't appear to be used *in validation*, but rather in `pub` functions of `OpType`: `other_port_count` and `port_index`....)

Happy to move these two renames into a separate PR if you like, I was going to when I thought there'd be 4 but I haven't bothered at the moment since it looks like there are only 2.